### PR TITLE
[add] Openbao single deployment

### PR DESCRIPTION
**Pull request by hmorris3293**

### CODE REVIEW

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+##Linode/SSH security settings
+#<UDF name="user_name" label="The limited sudo user to be created for the Linode: *No Capital Letters or Special Characters*">
+#<UDF name="disable_root" label="Disable root access over SSH?" oneOf="Yes,No" default="No">
+# <UDF name="client_ips" label="List of IP addresses to whitelist" example="192.168.1.2, 192.168.1.3" default="" />
+
+## Domain Settings
+#<UDF name="token_password" label="Your Linode API token. Required for Private IP check/add and DNS records [if applicable]">
+#<UDF name="subdomain" label="Subdomain" example="The subdomain for the DNS record: www (Requires Domain)" default="">
+#<UDF name="domain" label="Domain" example="The domain for the DNS record: example.com (Requires API token)" default="">
+#<UDF name="soa_email_address" label="email for SOA" default="">
```
Comment (n0vabyte) [inline deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:17]:
```suggestion
```
we don't use this variable anywhere in the playbook

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  fi
+
+  if [ "$DISABLE_ROOT" = "Yes" ]; then
+    echo "disable_root: yes" >> ${group_vars};
+  else echo "Leaving root login enabled";
+  fi
+  
+  if [[ -n ${SOA_EMAIL_ADDRESS} ]]; then
+    echo "soa_email_address: ${SOA_EMAIL_ADDRESS}" >> ${group_vars};
+  fi
```
Comment (n0vabyte) [inline deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:74]:
```suggestion
```
we don't use this variable anywhere in the playbook

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  for playbook in provision.yml site.yml; do ansible-playbook -v $playbook; done
+  
+}
+
+function installation_complete {
+  echo "Installation Complete"
+}
+# main
+run && installation_complete
+cleanup
```
Comment (n0vabyte) [inline deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:156]:
```suggestion
if [ "${DEBUG}" == "NO" ]; then
  cleanup
fi
```

File: apps/linode-marketplace-openbao/roles/openbao/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+## OpenBao is still in active development, so the openbao_download_link and the openbao_deb_file will need to actively be monitored for latest released here https://openbao.org/downloads/
+
+# global vars
+openbao_version: v2.0.0-beta20240618
+openbao_download_link: https://github.com/openbao/openbao/releases/download/v2.0.0-beta20240618/bao_2.0.0-beta20240618_linux_amd64.deb
+openbao_deb_file: /tmp/bao_2.0.0-beta20240618_linux_amd64.deb
+openbao_config: /etc/openbao/openbao.hcl
+openbao_ssl_path: /opt/openbao/tls
```
[the remark this answers is not in the document]
Comment (hmorris3293) [reply apps/linode-marketplace-openbao/roles/openbao/defaults/main.yml:9]: this is how the deb file configures it so leaving as is

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh (hunk 1 above)
Comment (hmorris3293) [reply deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:17]: this is for if the customer wants to create dns records

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh (hunk 2 above)
Comment (hmorris3293) [reply deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:74]: this is for if the customer wants to create dns records

File: deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+function configure_privateip {
+  LINODE_IP=$(get_privateip)
+  if [ ! -z "${LINODE_IP}" ]; then
+          echo "[info] Linode private IP present"
+  else
+          echo "[warn] No private IP found. Adding.."
+          add_privateip
+          LINODE_IP=$(get_privateip)
+          echo "Address=$LINODE_IP/17" | sudo tee -a /etc/systemd/network/05-eth0.network
```
Comment (n0vabyte) [inline deployment_scripts/linode-marketplace-openbao/openbao-deploy.sh:119]:
```suggestion
          echo "Address=$LINODE_IP/17" | tee -a /etc/systemd/network/05-eth0.network
```
You're already root

**Review by n0vabyte — APPROVED** [review]
deployment successful!

**Pull request merged**